tests/lib/rules/no-wait-for-empty-callback.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const rule = require('../../../lib/rules/no-wait-for-empty-callback');
+const RuleTester = require('eslint').RuleTester;
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({ parserOptions: { ecmaVersion: 2018 } });
+
+const ALL_WAIT_METHODS = ['waitFor', 'waitForElementToBeRemoved'];
+
+ruleTester.run('no-wait-for-empty-callback', rule, {
+  valid: [
+    ...ALL_WAIT_METHODS.map(m => ({
+      code: `${m}(() => {
+          screen.getByText(/submit/i)
+        })`,
+      errors: [
+        {
+          messageId: 'noWaitForEmptyCallback',
+        },
+      ],
+    })),
+    ...ALL_WAIT_METHODS.map(m => ({
+      code: `${m}(function() {
+          screen.getByText(/submit/i)
+        })`,
+      errors: [
+        {
+          messageId: 'noWaitForEmptyCallback',
+        },
+      ],
+    })),
+    {
+      code: `waitSomethingElse(() => {})`,
+    },
+  ],
+
+  invalid: [
```

Comment:
Would be nice to add more complex invalid cases like:

- callbacks receiving args but with empty body (e.g. `(foo, bar) => {}`)
- callbacks with body containing just comments (e.g. `() =>  { /* I'm empty anyway */ }`)
-  wait with no callback. What's the rule doing on this case? I'm sure it won't report anything but could cause an error in our rule code? Better to test that.

All of them for both default and arrow functions.